images/scripts/update-cilium-builder-image.sh

```diff
@@ -18,20 +18,17 @@ image="quay.io/cilium/cilium-builder"
 
 image_tag="$(WITHOUT_SUFFIX=1 "${script_dir}/make-image-tag.sh" images/builder)"
 
+image_full="${image}:${image_tag}"
+sha256=$("${script_dir}/get-image-digest.sh" "${image_full}" || echo "")
+if [ -n "${sha256}" ]; then
+  image_full="${image_full}@${sha256}"
+fi
+
 # shellcheck disable=SC2207
-used_by=($(git grep -l CILIUM_BUILDER_IMAGE= images/*/Dockerfile))
+used_by=($(git grep -l CILIUM_BUILDER_IMAGE= images/*/Dockerfile) "test/k8sT/manifests/test-verifier.yaml")
 
 for i in "${used_by[@]}" ; do
-  image_full="${image}:${image_tag}"
-  # Detect if the image_tag already exists, if it does then we can assume the
-  # image was created and a sha256 is available for it.
-  if grep "CILIUM_BUILDER_IMAGE=${image}:${image_tag}" "${i}" ; then
-    sha256=$("${script_dir}/get-image-digest.sh" "${image_full}")
-    if [ -n "${sha256}" ]; then
-      image_full="${image_full}@${sha256}"
-    fi
-  fi
-  sed "s|\(CILIUM_BUILDER_IMAGE=\)${image}:.*\$|\1${image_full}|" "${i}" > "${i}.sedtmp" && mv "${i}.sedtmp" "${i}"
+  sed -E "s#(CILIUM_BUILDER_IMAGE=|image: )${image}:.*\$#\1${image_full}#" "${i}" > "${i}.sedtmp" && mv "${i}.sedtmp" "${i}"
```

Comment:
Any particular reason for the temp file rather than using `sed -i`?

Reply:
I think this was originally added this way because `sed -i` on macOS works slightly different compared to GNU `sed` and this was the only portable way to implement the command.